### DISCUSSION TITLE
Forwarding proxy

### DISCUSSION THREAD

**Discussion by jimlloyd** [synchronizing/mitm#35] [General]
I am looking for a library to simplify the task of creating a service that accepts client connection for a given REST protocol, transforms the requests is some well-documented way, and then forwards the request to an existing server for the REST protocol. And of course handles the responses in the same fashion. I don't want to configure this as Web browser proxy, and I don't need to handle anything other than the REST protocol. And this can be for HTTP only, no need to do HTTPS/TLS MITM hacking.

My actual use case is to provide a service that appears much like ollama serving LLM chatbot clients. My application will be attached to a database and add Retrieval Augmented Generation capability. 

It's not clear to me that this `mitm` package can serve my use case. Or is there an API that is just not documented? If this package won't serve my use cause, can you recommend a good way to do this? I am aware of `requests` and `flask` which I may end up using but at the moment that solution seems somewhat awkward. I was hoping there would be a project like this one where I can just write some middleware.

**Comment by synchronizing** [comment]
I am a little confused by your question. REST is not a protocol, but rather an API architecture. What it seems like you are looking for is a normal Python HTTP server that takes in a request, queries a server/LLM model, and responds back to the client. This project would be a bit of an overkill for the sort of thing you are looking for, if I'm understanding you correctly. Flask/FastAPI should do the job just fine. If you describe your needs a bit more, I might be able to point you towards the right direction.
